frontend/src/components/TextEditor.jsx

```diff
@@ -24,12 +24,30 @@ export default function TextEditor({ document = [], onChange}) {
     initializeStateWithAllCommentThreads(editor, addCommentThread);
   }, [editor, addCommentThread]);
 
+  const onChange = (value) => {
+    updateDocumentState(value)
+    const isAstChange = editor.operations.some(
+      op => 'set_selection' !== op.type
+    )
+    if (isAstChange) {
+      // Save the value to Local Storage.
+      const content = JSON.stringify(value)
+      localStorage.setItem('content', content)
```

Comment:
I don't think we want to set it under just `content`. If you look at the `util/localDocStore.js` apis, we're storing it under a `{user_id: {file_id: file_object}}` structure so that's what we wanted to read/write save to. If we do it this way, then I think `content` will be overwritten each time you open a different file?

So ideally I think what you'd want to do is add a new `updateFile(file_id, file_obj)` function into the `localDocStore` file that accesses the file object store for the current user id and file id. Then it updates the file object that's saved there with the new state. and call that new function from here instead of `localStorage` directly.